core/templates/services/html-length.service.ts

```diff
@@ -108,15 +124,20 @@ export class HtmlLengthService {
 
   private getWeightForTextNodes(
       textNode: HTMLElement, calculationType: CalculationType): number {
-    const textContent = textNode.textContent || '';
+    const textContent = (textNode.textContent || '').trim();
```

Comment:
Why do you need the `|| ''`?

Reply:
Done! removed!

Reply:
null and the empty string are different things.

Please show an example for how .textContent can be null. At the very least there needs to be a descriptive comment here that explains why this can happen..

Reply:
OK, so if you drop the `|| ''`, what breaks, exactly?

Reply:
Sorry, that's not enough for me to suggest what to do. Shouldn't that be handled in the word count function if the issue is with splitting into an array?

Reply:
@seanlip I think is this solved with this [comment](https://github.com/oppia/oppia/pull/19518#discussion_r1489074927)?? Or any other changes is required? Please do let me know! Thanks!

Reply:
@seanlip I think this is solved with this https://github.com/oppia/oppia/pull/19518#discussion_r1489074927?? Or any other changes are required? Please do let me know! Thanks!

Reply:
Yes, I think an approach like that should help with this, thanks.

Reply:
Done!